### DISCUSSION TITLE
Statistics maintenance - PERSIST_SAMPLE_PERCENT

### DISCUSSION THREAD

**Discussion by GravierPL** [olahallengren/sql-server-maintenance-solution#518] [Questions & Answers]
Hello There,
Recently for one of my database table I need to set PERSIST_SAMPLE_PERCENT to 100 (fullscan) for couple of indexex in order to force automatic stats update to use fullscan. This is being reset with index rebuild. I looked here (https://ola.hallengren.com/sql-server-index-and-statistics-maintenance.html) but I did not found any option to enable this with Ola's scripts so I can make this part of our daily routine.
Am I missing something?

**Comment by GeorgePalacios** [comment]
You are looking for the "StatisticsResample" option:

https://ola.hallengren.com/sql-server-index-and-statistics-maintenance.html#StatisticsResample

**Reply by GravierPL** [reply]
Thanks for reply, but I think this is not that. I just run test and StatisticsResample cannot be combined with StatisticsSample, so fullscan was not performed nor PERSIST_SAMPLE_PERCENT was enabled for stats on tested table.

**Comment by GeorgePalacios** [comment]
If you run an update of those stats with FULLSCAN and
PERSIST_SAMPLE_PERCENT on (manually) - that will set the sample rate to
100. If you then use RESAMPLE option with Olas solution, it will pick up
that previously set 100% for those statistics moving forward. We have this
running in our environment atm.

**Reply by GravierPL** [reply]
That is correct, but not fit for mu purpose, as on of application uses their own job to rebuild indexes. That resets PERSIST_SAMPLE_PERCENT, so I was hoping that I can re-enable it during nightly run of Ola's scripts. But it seems I will need to use my own code for that.

Thanks!
